src/RED/components/Power&BMS.tsx

```diff
@@ -103,83 +102,95 @@ interface IState {
+
+    rovecomm.on("PackI_Meas", (data: number[]) => this.batteryListenHandler(data, "PackI_Meas"))
+    rovecomm.on("PackV_Meas", (data: number[]) => this.batteryListenHandler(data, "PackV_Meas"))
+    rovecomm.on("Temp_Meas", (data: number[]) => this.batteryListenHandler(data, "Temp_Meas"))
+    rovecomm.on("CellV_Meas", (data: number[]) => this.batteryListenHandler(data, "CellV_Meas"))
+    console.log(boardTelemetry)
   }
 
-  boardListenHandler(data: number[], partList: string[], isEnabler: boolean): void {
+  /**
+   * @desc takes data from rovecomm and applies those values to the state object
+   * all rovecomm telemetry values are transported as arrays, even if only a single value
+   * is transferred. IF a single value is transferred, the data is assumed to be a bitmask,
+   * after which it is unpacked and has the resultant boolean values applied to the 'enabled'
+   * properties of each component object. If there is more than one value, the data is assumed
+   * to be the list of amperage floats, and assigned to the 'value' property.
+   * @param data contains either a single bitmask number or a list of numbers; both inside of an array
+   * @param partList the list of components determined by the state object
+   */
+  boardListenHandler(data: number[], partList: string): void {
     const { boardTelemetry } = this.state
-    if (isEnabler) {
-      const bitmask = BitmaskUnpack(data[0], partList.length)
-      for (let i = 0; i < partList.length; i++) {
-        boardTelemetry[partList[i]].enabled = Boolean(Number(bitmask[i]))
-      }
+    if (data.length > 1) {
+      boardTelemetry[partList].forEach((part: string, index: number) => {
+        boardTelemetry[partList][part].value = data[index]
+      })
     } else {
-      for (let i = 0; i < partList.length; i++) {
-        boardTelemetry[partList[i]].value = data[i]
-      }
+      const bitmask = BitmaskUnpack(data[0], partList.length)
+      boardTelemetry[partList].forEach((part: string, index: number) => {
+        boardTelemetry[partList][part].enabled = Boolean(Number(bitmask[index]))
+      })
     }
     // The setState is kept until the end because of how the priority of state changes are handled.
     // This reason remains the same for all the functions with a setState at the end.
     this.setState({ boardTelemetry })
   }
 
-  batteryListenHandler(data: number, partList: string[]): void {
+  batteryListenHandler(data: number[], part: string): void {
     const { batteryTelemetry } = this.state
-    for (let i = 0; i < partList.length; i++) {
-      batteryTelemetry[partList[i]].value = data[i]
+    if (data.length > 1) {
+      batteryTelemetry[part].forEach((cell: any, index: number) => {
+        batteryTelemetry[part][cell].value = data[index]
+      })
+    } else {
+      batteryTelemetry[part].value = data[0]
     }
     this.setState({ batteryTelemetry })
   }
```

Comment:
This one doesn't have as much of a need for separation. Where the Board objects needed two separate properties set, the only Battery object that needs special logic are the list of cells. It'd be especially annoying since I'm also adding proper documentation this time